ribadstore.cc

```diff
@@ -152,6 +152,51 @@ namespace ns3
         NS_LOG_INFO("Send status: " << socket->SendTo(p, 0, dest));
     }
 
+
+    /* Return true if the provided entry is in db, false if table not updated */
+    bool
+    RIBAdStore::UpdateNameCache(NameDBEntry* advertised)
```

Comment:
See if the name store needs update with the received advertisement. Return value is set correspondingly to reflect that.

Reply:
Current problem with this implementation is that if the DC owner changes the name's r_transitivity, there is no way for peered ASes to figure out when to overwrite. 

We might need more attributes in the advertisement packet scheme.